src/main/java/com/jay/shoppingmall/service/handler/FileHandler.java

```diff
@@ -19,112 +22,100 @@
-                    if (contentType.contains("image/jpeg"))
-                        originalFileExtension = ".jpg";
-                    else if (contentType.contains("image/png"))
-                        originalFileExtension = ".png";
-
-                String fileName = System.nanoTime() + "";
-                        //확장자를 붙여 저장하는 것은 보안에 취약.
-                        //+ originalFileExtension;
-
-                Image image = Image.builder()
-                        .originalFileName(multipartFile.getOriginalFilename())
-                        .fileSize(multipartFile.getSize())
-                        .filePath(path + File.separator + fileName)
-                        .fileExtension(originalFileExtension)
-                        .imageRelation(imageRelation)
-                        .foreignId(foreignId)
-                        .isMainImage(false)
-                        .build();
-
-                //업로드 한 파일 데이터를 지정한 파일에 저장.
-                file = new File(absolutePath + path + File.separator + fileName);
-                try {
-                    multipartFile.transferTo(file);
+        //파일 저장 세부 경로 설정.
+        String path = "images" + File.separator + currentDate;
+        File file = new File(path);
+        File thumbnailFile = new File(path + File.separator + "thumbnails");
+
+        //폴더가 존재하지 않으면 만듬
+        //없는 폴더에 파일을 넣으려고 하면 I/O 익셉션 발생
+        if (!file.exists()) {
+            file.mkdirs();
+        }
+        if (!thumbnailFile.exists())
+            thumbnailFile.mkdirs();
+
+        //모든 파일에 대해 처리
+        //파일의 확장자 추출
+        String originalFileExtension = "";
+        String contentType = multipartFile.getContentType();
+
+        //확장자가 없다면 처리 중지
+        if (contentType.contains("image/jpeg"))
+            originalFileExtension = ".jpg";
+        else if (contentType.contains("image/png"))
+            originalFileExtension = ".png";
+
+        String fileName = System.nanoTime() + "";
+        //확장자를 붙여 저장하는 것은 보안에 취약.
+        //+ originalFileExtension;
+
+        Image image = Image.builder()
+                .originalFileName(multipartFile.getOriginalFilename())
+                .fileSize(multipartFile.getSize())
+                .filePath(path + File.separator + fileName)
+                .fileExtension(originalFileExtension)
+                .imageRelation(imageRelation)
+                .foreignId(foreignId)
+//                .isMainImage(false)
+                .build();
+
+        //업로드 한 파일 데이터를 지정한 파일에 저장.
+        file = new File(absolutePath + path + File.separator + fileName);
+        try {
+            multipartFile.transferTo(file);
 //                    Thumbnails.of(file)
 //                            .size(750, 500)
 //                            .toFile(thumbnailFile + File.separator + fileName);
-                } catch (IOException e) {
-                    throw new FileException("File I/O failed");
-                }
+        } catch (IOException e) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)